extensions/account-registration.md

```diff
@@ -0,0 +1,187 @@
+---
+title: `account-registration` Extension
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Ed Kellett"
+    period: "2020"
+    email: "e@kellett.im"
+  -
+    name: "Val Lorentz"
+    period: "2021"
+    email: "progval+ircv3@progval.net"
+---
+
+
+## Motivation
+
+The lack of a standardized way to register an account has been a source
+of frustration for many participants for several years.
+
+This isn't the first attempt to fix it. Unlike the withdrawn `ACC` proposal,
+this effort aims to be more appealing to implementors by being severely
+restricted in scope.
+The only register-verify flows supported are those already in use in the wild.
+The authors hope that this stripped-down solution will serve as the basis
+for future organic expansion.
+
+### Capability
+
+This specification adds the `draft/register` capability, whose presence
+signifies that the server accepts the `REGISTER` command.
+
+The capability has an optional value, a comma-separated list of key-value
+pairs; the format is intended to follow the precedent set by
+[the multiline draft][multiline].
+
+The defined keys are:
+
+ * `before-connect` - if present, indicates the server supports early
+   registration, so it MUST NOT use
+   `FAIL REGISTER COMPLETE_CONNECTION_REQUIRED`
+ * `email-required` - if present, registrations require a valid email address
+   to process
+
+Clients MUST ignore any value assigned to these keys, and MUST ignore
+any unknown key.
+
+Software implementing this work-in-progress specification MUST NOT use
+the unprefixed `register` or `account-registration` capability names.
+Instead, implementations SHOULD use the `draft/register`
+capability name to be interoperable with other software implementing
+a compatible work-in-progress version.
+
+### Commands
+
+    REGISTER {<email> | "*"} <password>
+    
+The `REGISTER` command informs the server of a request to register
+an account named for the current nick of the requestor.
+
+The `REGISTER` command MAY be sent at any point during the connection
+that the client has a valid nickname.
+If the client sends `REGISTER` before completing connection registration,
+and receives a `FAIL REGISTER COMPLETE_CONNECTION_REQUIRED`, it SHOULD make
+a second attempt after it receives the welcome message.
+
+    VERIFY <account> <code>
+    
+The `VERIFY` command completes a registration that required email verification.
```

Comment:
Not all environments want to send email, for example, Ophion intends to prompt the user to complete a CAPTCHA it generates at some URI.  Changing "email verification" to "verification" would allow more flexibility.